src/scripts/main.js

```diff
@@ -1,3 +1,210 @@
 'use strict';
 
-// write your code here
+const columns = [[], [], [], []];
+const rows = Array.from(document.querySelectorAll('.field-row'));
+const cells = document.querySelectorAll('.field-cell');
+const cellsInRow = rows.map(row => [...row.querySelectorAll('.field-cell')]);
+
+for (let i = 0; i < 4; i++) {
+  for (let j = 0; j < 4; j++) {
+    columns[j].push(rows[i].children[j]);
+  }
+}
+
+const button = document.querySelector('.start');
+const startMessage = document.querySelector('.message-start');
+const loseMessage = document.querySelector('.message-lose');
+const winMessage = document.querySelector('.message-win');
+const score = document.querySelector('.game-score');
+const moveSound = document.getElementById('audio-move');
+const looseSound = document.getElementById('audio-loose');
+const winSound = document.getElementById('audio-win');
+
+let clearCells = [...cells];
+
+function getRandomCell() {
+  return clearCells[Math.floor(Math.random() * clearCells.length)];
+}
+
+function getCellValue() {
+  return Math.random() < 0.9 ? 2 : 4;
+}
+
+function newCell() {
```

Comment:
The condition 'clearCells.length === undefined' can be simplified to '!clearCells.length' which checks if the array is empty or not.